src/common/WindowDescriptors.cpp

```diff
@@ -110,6 +110,42 @@ void SplitDescriptor::loadFromJSON(SplitDescriptor &descriptor,
     descriptor.filters_ = loadFilters(root.value("filters"));
 }
 
+TabDescriptor TabDescriptor::loadFromJSON(const QJsonObject &tabObj)
+{
+    TabDescriptor tab;
+    // Load tab custom title
+    QJsonValue title_val = tabObj.value("title");
```

Comment:
Please use camelCase for vairable names. I know this variable name was named like this before as well, but since you're refactoring this might be changed as well.